cmd/deployment/metadata.go

```diff
@@ -0,0 +1,72 @@
+package project_map
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/go-git/go-git/v5"
+	errors "github.com/pkg/errors"
+	"github.com/urfave/cli/v2"
+)
+
+type Metadata struct {
+	Deployment Item `json:"deployment"`
+}
+
+type Item struct {
+	Sha        string `json:"sha"`
```

Comment:
I struggled with simplifying this into a single custom struct `Deployment` as a `map[string]string` and compiling the string in a 'clean' way. This approach made more sense/was cleaner _but_ I've got a hunch it's inefficient/overly verbose?

Reply:
Nope this is pretty much perfect. `map[string]string` works best for single-level objects with dynamic attributes. In this case we know what data we want to capture from git and render into the metadata object.